Baikal/PostEffects/ML/denoiser.h

```diff
@@ -14,12 +44,46 @@ namespace Baikal
         enum class MLDenoiserInputs
         {
             kColorDepthNormalGloss7,
+            kColorAlbedoNormal8,
         };
 
-        std::unique_ptr<Inference> CreateMLDenoiser(MLDenoiserInputs inputs,
-                                                    float gpu_memory_fraction,
-                                                    std::string const& visible_devices,
-                                                    std::size_t width,
-                                                    std::size_t height);
+        class MLDenoiser : public PostEffect
+        {
+        public:
+
+            MLDenoiser(const CLWContext& context, std::size_t width, std::size_t height);
+
+            InputTypes GetInputTypes() const override;
+
+            void Apply(InputSet const& input_set, Output& output) override;
+
+            void Update(Camera* camera, unsigned int samples) override;
```

Comment:
This method passes current samples count and camera to particular PostEffect implementation, and PostEffect applies them and calculate needed parameters:
1) MLDenoiser uses sample count to control input pipeline, because this pipeline management is part of PostEffect
2) BilateralDenoiser uses current sample count to calculate internal paramters
2) WaveletDenoiser also uses camera to calculate internal parameters.

Method Update(camera, samples) allows client to not downcast PostEffect to concrete implementation and incapsulates internal parameters calculation inside particular PostEffect. Therefore, client code becomes simpler.